spec/Subject-spec.ts

```diff
@@ -683,6 +683,30 @@ describe('Subject', () => {
       expect(true).to.be.true;
     });
   });
+
+  describe('many subscribers', () => {
+    it('should be able to subscribe and unsubscribe huge amounts of subscribers', () => {
+      let numResultsReceived = 0;
+      const allSubscriptions: Subscription[] = [];
+      const source = new Subject<number>();
+      const numSubscribers = 100000;
```

Comment:
With 100k items, the version of the code currently on master exceeds the timeout of mocha on a GitHub Codespaces (it reached 10s contrary to new code taking around 400ms). But I don't really know which bar I should put their as the more powerful runners will be the more unlikely the failure even if we reintroduce a slow version. Maybe you have suggestion of it 🤔